pkg/provisioner/ironic/ironic.go

```diff
@@ -321,6 +321,52 @@ func (p *ironicProvisioner) getNode() (*nodes.Node, error) {
 	}
 }
 
+// Verifies that node has port assigned by Ironic.
+func (p *ironicProvisioner) nodeHasAssignedPort(ironicNode *nodes.Node) bool {
+	opts := ports.ListOpts{
+		Fields: []string{"node_uuid"},
+	}
+
+	opts.NodeUUID = ironicNode.UUID
+
+	pager := ports.List(p.client, opts)
+	if pager.Err != nil {
+		p.log.Info("node has no assigned port", "error", pager.Err)
+		return false
+	}
+
+	allPages, err := pager.AllPages()
+	if err != nil {
+		p.log.Info("node has no assigned port", "error", err)
+		return false
```

Comment:
Is this the right way to handle errors? I'd have thought we'd want to return the error and retry.

Reply:
I'd like to keep signature of parent method as simple as possible, with only `bool` in returned values
( `(r *receiver) f(a args) bool` ) so it can be used if `if` conditions as simple as possible.

For using retries in that case one would have to decide how many times to retry and implement some kind of backoff.

Reply:
> I'd like to keep signature of parent method as simple as possible

Ignoring errors makes it simpler than is possible :)

> For using retries in that case one would have to decide how many times to retry and implement some kind of backoff.

Just `return transientError(err)`. That's what it's for.

Reply:
returning error would require to change logic in 
```
if p.bootMACAddress != "" && !p.nodeHasAssignedPort(ironicNode) && !p.isPortAllocatedToAddress(p.bootMACAddress) {
```

to some nested `if`

We will lose readability because we need to differentiate `error` from `bool`, so returned values from function will have to be `(error, bool)`

Will do if this (`(error, bool)`) sound as acceptable solution.

Reply:
> We will lose readability because we need to differentiate `error` from `bool`,

correctness > readability

> so returned values from function will have to be `(error, bool)`
> 
> Will do if this (`(error, bool)`) sound as acceptable solution.

`bool, error` is idiomatic in go, but yes.